Lib/posixpath.py

```diff
@@ -216,6 +216,12 @@ def ismount(path):
     return False
 
 
+def isreserved(path):
+    """Return true if the pathname is reserved by the system."""
+    os.fspath(path)
```

Comment:
Why this call if you're not going to use the result?

```suggestion
```

Reply:
To ensure an exception is raised if the given argument isn't `PathLike`. If that sounds like a good idea I can add a comment explaining, otherwise I'll accept your suggestion!

Reply:
Can't find this elsewhere in the standard library actually, so will accept your suggestion.

Reply:
What do you mean in terms of not being able to find it? The implementation of `isabs()` calls `os.fspath()`, and it does not ignore a `TypeError`. Others such as `isdir()`, `isfile()`, `islink()`, and `ismount()` also fail by way of `os.lstat()` or `os.stat()` with a `TypeError` if the value isn't `PathLike` or an integer (file descriptor).

Reply:
In those cases, the return value of `os.fspath()` is used for something. I can't find an example that calls `fspath()` (even indirectly) *only* to check the argument type.

Reply:
I agree with your initial response last week. For example, if `isreserved(1.2)` raises `TypeError` in Windows, it should as well in POSIX.

Reply:
Tentatively agree, but lets see what @brettcannon says :)

Reply:
When I brought in `os.fspath()`, it was added to the appropriate points where conversion was necessary, otherwise paths were left alone to trigger where conversion was critical. If this is to match the exception that happens on Windows, that makes sense, but a comment is needed to avoid future confusion. 😅

Reply:
Thanks both, I've restored the `os.fspath()` call and added a comment.